controllers/storageclient_controller.go

```diff
@@ -549,6 +561,32 @@ func (r *StorageClientReconciler) reconcileSharedfileStorageClaim() error {
 	return nil
 }
 
+func (r *StorageClientReconciler) reconcileRemoteNoobaa() error {
+	joinSecret := &corev1.Secret{}
+	joinSecret.Name = joinSecretName
+	joinSecret.Namespace = r.OperatorNamespace
+	if err := r.get(joinSecret); err != nil {
+		r.Log.Error(err, "failed to get the joinSecret for remote noobaa")
+		return err
+	}
```

Comment:
This is unnecessary.  Getting to this part of the code means that you already called GetConfig from the provider, which guarantees a joint secret was created. 

Let's remove this check

Reply:
Removed fetching joined secret again